keps/sig-windows/0000-20181221-windows-group-managed-service-accounts-for-container-identity.md

```diff
@@ -110,31 +110,76 @@ When this app moves to production on containers, the team will need to coordinat
   - Provision the GMSA and gives details to Kubernetes cluster admin.
   - Assign access to a AD security group to control what machines can use this GMSA. The AD security group should include all authorized Kubernetes nodes.
 
-2. A Kubernetes cluster admin (or cluster setup tools or an operator) will install a cluster scoped CRD (of kind GMSACredSpec) whose instances will be used to store GMSA credential spec configuration:
+2. A Kubernetes cluster admin (or cluster setup tools or an operator) will install a cluster scoped CRD (of kind GMSACredentialSpec) whose instances will be used to store GMSA credential spec configuration:
 
 ```
 apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
-  name: gmsacredspecs.auth.k8s.io
+  name: gmsacredentialspecs.windows.k8s.io
 spec:
-  group: auth.k8s.io
+  group: windows.k8s.io
   version: v1alpha1
   names:
-    kind: GMSACredSpec
-    plural: gmsacredspecs
+    kind: GMSACredentialSpec
+    plural: gmsacredentialspecs
   scope: Cluster
+  validation:
+    openAPIV3Schema:
+      properties:
+        credspec:
+          description: GMSA Credential Spec
+          properties:
+            ActiveDirectoryConfig:
+              description: Windows Active Directory configuration
+              properties:
+                GroupManagedServiceAccounts:
+                  description: GMSA names and scopes
+                  type: array
```

Comment:
i am thinking a little bit more about this. a single container can only have a single gMSA associated with it. why is the GroupManagedServiceAccounts an array? it should only have a single gMSA

Reply:
The reference script `New-CredentialSpec` at https://raw.githubusercontent.com/Microsoft/Virtualization-Documentation/live/windows-server-container-tools/ServiceAccounts/CredentialSpec.psm1 allows you to specify `AdditionalAccounts`. The primary GMSA + the additional accounts end up in the array. Also note that the primary GMSA will always have two entries: one with scope DNSRoot, the other with scope NetBIOSName as generated by the reference script.